src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -407,40 +504,52 @@ class Definitions {
     //  - .linkedClass: the ClassSymbol of the enumeration (class E)
     sym.owner.linkedClass.typeRef
 
-  object FunctionType {
+  object FunctionOf {
```

Comment:
Could we please maintain the old name `FunctionType`? 
I do not think that new name is substantially better to justify source incompatibility with various unmerged branches.